control-plane/p4RuntimeArchStandard.cpp

```diff
@@ -0,0 +1,715 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <set>
+#include <unordered_map>
+
+#include <boost/optional.hpp>
+
+#include "frontends/common/resolveReferences/referenceMap.h"
+#include "frontends/p4/fromv1.0/v1model.h"
+#include "frontends/p4/typeMap.h"
+#include "ir/ir.h"
+#include "lib/log.h"
+#include "typeSpecConverter.h"
+
+#include "p4RuntimeArchHandler.h"
+
+using ::P4::ControlPlaneAPI::Helpers::getExternInstanceFromProperty;
+using ::P4::ControlPlaneAPI::Helpers::addAnnotations;
+
+namespace p4configv1 = ::p4::config::v1;
+
+namespace P4 {
+
+/** \addtogroup control_plane
+ *  @{
+ */
+namespace ControlPlaneAPI {
+
+namespace Standard {
```

Comment:
Interesting choice to call this namespace "Standard". I expect that indeed these will be part of the standard library of externs. However, you also chose to have v1model part of this file, and in fact part of the Standard namespace. Why this choice?

Reply:
For me v1model + PSA constitute the standard architectures, and I was going to put all the code specific to these architectures under `Standard`. I don't have support for PSA but I expect I will try to re-use as much code as possible.